src/components/App.tsx

```diff
@@ -163,14 +164,40 @@ function App() {
   const [nodes, setNodes, onNodesChange] = useNodesState([]);
   const [edges, setEdges, onEdgesChange] = useEdgesState([]);
 
+  const edgeUpdateSuccessful = useRef(true);
+
+  const onEdgeUpdateStart = useCallback(() => {
+    edgeUpdateSuccessful.current = false;
+  }, []);
+
+  const onEdgeUpdate = useCallback((oldEdge: Edge<any>, newConnection: Connection) => {
+    if (getFluxNodeParent(nodes, edges, newConnection.target!) !== undefined) {
+      return;
+    }
+
+    edgeUpdateSuccessful.current = true;
+    setEdges((edges) => updateEdge(oldEdge, newConnection, edges));
+  }, [nodes, edges]);
+
+  const onEdgeUpdateEnd = useCallback((_: unknown, edge: Edge<any>) => {
+    if (!edgeUpdateSuccessful.current) {
+      setEdges((edges) => edges.filter((e) => e.id !== edge.id));
+    }
+
+    edgeUpdateSuccessful.current = true;
+  }, []);
+
```

Comment:
wait sorry why do we need these extra callbacks? returning in `onConnect` accomplishes what we need no?

otherwise lgtm

Reply:
oh i see this is to add edge manipulation?

Reply:
> oh i see this is to add edge manipulation?

yes, not sure this is something you want but thought it makes sense now that there is only one parent enforced.

Reply:
yeah this looks great actually— nice job. in the future would prefer if these were two PRs as this jumbles a bit of complexity but no worries this time

Reply:
Makes sense, will remember that :)